wgpu-hal/src/metal/adapter.rs

```diff
@@ -844,6 +844,12 @@ impl super::PrivateCapabilities {
                 && ((device.supports_family(MTLGPUFamily::Apple8)
                     && device.supports_family(MTLGPUFamily::Mac2))
                     || device.supports_family(MTLGPUFamily::Apple9)),
+            // https://developer.apple.com/metal/Metal-Feature-Set-Tables.pdf#page=6
+            float_atomics: family_check
+                && (device.supports_family(MTLGPUFamily::Apple7)
+                    || device.supports_family(MTLGPUFamily::Apple8)
+                    || device.supports_family(MTLGPUFamily::Apple9)
+                    || device.supports_family(MTLGPUFamily::Mac2)),
```

Comment:
```suggestion
                && (device.supports_family(MTLGPUFamily::Apple7)
                    || device.supports_family(MTLGPUFamily::Mac2)),
```

Only checking for 7 is enough since the higher families support 7 as well.